azurerm/internal/services/timeseriesinsights/tests/resource_time_series_insights_environment_test.go

```diff
@@ -0,0 +1,167 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/timeseriesinsights/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMTimeSeriesInsightsEnvironment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_time_series_insights_environment", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMTimeSeriesInsightsEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMTimeSeriesInsightsEnvironment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMTimeSeriesInsightsEnvironmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMTimeSeriesInsightsEnvironment_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_time_series_insights_environment", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMTimeSeriesInsightsEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMTimeSeriesInsightsEnvironment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMTimeSeriesInsightsEnvironmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMTimeSeriesInsightsEnvironment_complete(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMTimeSeriesInsightsEnvironmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMTimeSeriesInsightsEnvironment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMTimeSeriesInsightsEnvironmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
```

Comment:
Does this update test really help? Because in the schema, the `tags` are defined as `ForceNew`, and the config for basic does not have a tag, complete config has a tag. Therefore this update would always be a replacement rather than a in-place update.